### DISCUSSION TITLE
Indexing Causes Container To Be Killed 💀

### DISCUSSION THREAD

**Discussion by P-Verbrugge** [photoprism/photoprism#1714] [Q&A] [answered]
**What does not work as expected?**

When importing "originals", facenet is loaded. It seems that facenet is not loading properly or doing something and then crashes:

> time="2021-11-09T18:45:53+01:00" level=info msg="faces: loading facenet"
> /entrypoint.sh: line 94:     9 Killed                  "$@"

After the crash it restarts

**How can we reproduce it?**

1. Install PhotoPrism on Raspberry Pi OS (on a pi 4) with docker. Use container: photoprism/photoprism:latest
2. Upload folders to your drive
3. Login into PhotoPrism. 
4. Go to library -> Orignals
5. Start the import and watch the logs. Crash happens after facenet loading.

**What behavior do you expect?**

I expected the originals to be imported

**What version you are using?**

PhotoPrism 211018-e200f322-Linux-armv7

**Comment by P-Verbrugge** [comment, marked as answer]
My Pi has 4GB ram. I have not set the swap file. Not sure how to do that.

**Reply by lastzero** [reply]
Please run these commands in a terminal if you have no swap configured yet:

```
sudo -i
fallocate -l 4G /swapfile
chmod 600 /swapfile
mkswap /swapfile
swapon /swapfile
echo '/swapfile none swap sw 0 0' | tee -a /etc/fstab
```

You're welcome!

**Reply by P-Verbrugge** [reply]
Thanks. Will give this a try later.

**Reply by lastzero** [reply]
Follow the link in our docs on https://docs.photoprism.org/getting-started/raspberry-pi/ to learn more about swap:
https://opensource.com/article/18/9/swap-space-linux-systems

> Your device should have at least 4 GB of memory. Also make sure it has at least 4 GB of swap configured, so that indexing doesn't cause restarts when there are memory usage spikes.

**Reply by lastzero** [reply]
See also:
- https://github.com/photoprism/photoprism/issues/1390
- https://github.com/photoprism/photoprism/issues/1503
- https://github.com/photoprism/photoprism/issues/1094
- https://github.com/photoprism/photoprism/issues/900
- https://github.com/photoprism/photoprism/issues/876
- https://github.com/photoprism/photoprism/issues/762

**Reply by P-Verbrugge** [reply]
Thanks. I have my test setup running. It seems stable. Hopefully my production machine will do the same. :-)

**Reply by lastzero** [reply]
Glad to hear that! See https://docs.photoprism.org/funding/ for how to support our work.

**Reply by lastzero** [reply]
Added this to our Troubleshooting Guide: https://docs.photoprism.org/getting-started/troubleshooting/#adding-swap
